### ISSUE TITLE
Sync without Dropbox or RemoteStorage?

### ISSUE THREAD

**Issue by herold2k** [Laverna/laverna#635]
Hi there,

first of all thanks for the great work - Laverna is a really cool tool!
With that said, I have some questions/wishes regarding Laverna:

1. I do not have a Dropbox- or RemoteStorage-Account. Is it possible to sync Laverna notes between devices (Linux, Windows, Android) using another way? Maybe with a self-hosted Laverna installation or via personal webspace (FTP, SSH, WebDAV...)?
Something like: [laverna_desktop] <-> [personal_webspace] <-> [laverna_android]

2. Is Laverna secure enough resp. is the encryption strong enough that I _could_ host it anywhere and _could_ sync it via cloud services like Dropbox or RS, even if I don't trust them in terms of privacy?

Thanks!

**Comment by userdaphi** [comment]
Yeah, would be cool to connect it with yourr own ownCloud or better Nextcloud. I won't use Dropbox and RemoteStorage. WebDAV over https would be great.

**Comment by eriteric** [comment]
Nextcloud now has an "app store", adding laverna there would be pretty neat

**Comment by romanxbayramov** [comment]
Currently, you can sync Laverna through GDrive, OneDrive, MEGA, etc. with FS module.

1. Enable the module in Settings -> Modules -> FS
2. Click FS -> Settings and select your desired cloud folder on your system.
![image](https://cloud.githubusercontent.com/assets/22922027/19496401/07cf2708-9590-11e6-9e60-9020edd5f525.png)
3. Profit.

![image](https://cloud.githubusercontent.com/assets/22922027/19496441/31a322f0-9590-11e6-855e-5748bbbd650b.png)

**Comment by herold2k** [comment]
But this is exactly what I do NOT want, when using Laverna: rely on other companies like Dropbox, Google, Microsoft, MEGA etc.

I want to use my OWN webspace/server/... and this should as simple and straightforward as possible.

Otherwise I could just use Google Keep or MS OneNote etc.

**Comment by r0bbie** [comment]
Remain really hopeful for this via OwnCloud/NextCloud/WebDav support (#155). It's currently a pretty critical missing feature (for anyone who doesn't use Dropbox).

**Comment by berzas** [comment]
Would be great to sync trough Webdav, Laverna is perfect except of that.

**Comment by jekuno** [comment]
Would it be too hard to run your own remote storage server? There are several open source options: https://wiki.remotestorage.io/Servers#Host_your_own

You could even host RemoteStorage on your own server and keep it behind the firewall of your local intranet. Then you could sync Laverna when you are at home (and WiFi-connect your smartphone or desktop to the same network as your local server).

**Comment by herold2k** [comment]
Yep, running my own remote storage server would be 'too hard'. Since there seems to be no way to host a remote storage server on shared webspace and I don't want to run my own server just because of remote storage... So this is no choice.

**Comment by funilrys** [comment]
I mean webdav is just what we need !! ;)

**Comment by smu55** [comment]
Lots of people use syncthing or relisio sync (former bittorent sync) to keep things synchronised. It would be nice If one could sync notes locally this way.

**Comment by jancborchardt** [comment]
[Nextcloud](https://nextcloud.com) can be fully self-hosted so you’re not relying on any other service. :) We also support simple WebDAV as API for the files. :) https://github.com/Laverna/laverna/issues/104

The cool folks of Buttercup recently added Nextcloud integration too: https://github.com/buttercup/buttercup-core/pull/151 – maybe @perry-mitchell @sallar @brantje can provide some pointers on how they did that? :)

**Comment by perry-mitchell** [comment]
WebDAV is an interesting protocol, and it has a lot of quirks in my opinion. The WebDAV engine Nextcloud uses (I believe it's SabreDAV?) is quite alright but many implementations get a bit loose when it comes to response structures. I wrote webdav-client and webdav-fs to handle several styles of WebDAV servers, including the one Nextcloud uses. WebDAV-client uses promises to work with the files on the server, making reading and writing a piece of cake. WebDAV is a pretty solid protocol with a ton of support and compatible libraries 👍

**Comment by jancborchardt** [comment]
@perry-mitchell would you be down to write a blog post on how other web apps like Buttercup (and Laverna in this case) can use your library to use Nextcloud as storage? :)

Would be pretty awesome, and we would be happy to publish it on our blog and promote it. :) cc @jospoortvliet

**Comment by perry-mitchell** [comment]
I'd be very keen to write a post.. let's sync up on that via email (perry at perrymitchell.net)

**Comment by coloradomountain** [comment]
Surprised there is no STFTP support and all of this love for WEBDAV. Will fork and add SFTP support.

**Comment by domacikolaci** [comment]
@coloradomountain how did you go with your own fork? Has anyone made their own version of this with WebDAV or something? Really want to use this with nextcloud. Thanks!

**Comment by coloradomountain** [comment]
@domacikolaci I never forked Laverna. I simply started my own project. I am unsure the scope of Laverna was ever remote storage of notes, but integrating it shouldn't be an issue, I just never found the need.

**Comment by domacikolaci** [comment]
@coloradomountain ahhhh bugger, all good. Wish I knew the language, would attempt it. Know any APIs or something to go off? Might as well give it a crack if noone else will :)

**Comment by coloradomountain** [comment]
@domacikolaci  

https://www.npmjs.com/package/webdav
https://github.com/Laverna/laverna#coding-style-guidelines

Really should be simple. Best of luck!

**Comment by domacikolaci** [comment]
Thanks @coloradomountain will check it out!

**Comment by jancborchardt** [comment]
There's this great blog post by @perry-mitchell on how to conmect apps to Nextcloud using his webdav-fs library:
https://nextcloud.com/blog/using-webdav-fs-to-access-files-in-nextcloud/

Probably quite helpful to people who want to improve Laverna by these additional sync capabilities. :)

**Comment by domacikolaci** [comment]
Cool, will look into. In the end I'll be honest, got lazy and looked into Nextcloud notes + using QOwnNotes which is working really well right now for me. Will look into getting Laverna working instead, Laverna still looks like the better option in the end, when I understand JS more, but for now, it's all going over my head :D

**Comment by Astronach** [comment]
With the module 'FS' you can "sync" the data with a specific folder on your file system. This folder can be sync with syncing

**Comment by funilrys** [comment]
Hello there,
Please report to https://github.com/Laverna/laverna/issues/971#issuecomment-411423965 which explain the state of this project.

Have a nice day/night,
Cheers,
Nissar
